File: src/main/java/org/jetbrains/fortran/lang/psi/FortranLabeledDoConstruct.java
package org.jetbrains.fortran.lang.psi;

import java.util.List;

import org.jetbrains.annotations.*;

public interface FortranLabeledDoConstruct extends FortranExecutableConstruct {

    @NotNull
    List<FortranExpr> getExprList();

    @NotNull
    FortranLabelDoStmt getLabelDoStmt ();

    @Nullable
    FortranStmt getDoTermActionStmt();

    @Nullable
    FortranLabeledDoConstruct getLabeledDoTermConstract();

    @Nullable
    FortranEndDoStmt getEndDoStmt();
}

File: src/main/kotlin/org/jetbrains/fortran/lang/psi/FortranLabeledDoConstruct.kt
package org.jetbrains.fortran.lang.psi

interface FortranLabeledDoConstruct : FortranExecutableConstruct {
    val exprList: List<FortranExpr?>
    val labelDoStmt: FortranLabelDoStmt
    val doTermActionStmt: FortranStmt?
    val labeledDoTermConstract: FortranLabeledDoConstruct?
    val endDoStmt: FortranEndDoStmt?
}
